metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/window/WindowProcessor.java

```diff
@@ -0,0 +1,321 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.apache.metron.profiler.client.window;
+
+import org.antlr.v4.runtime.*;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.apache.metron.common.dsl.ErrorListener;
+import org.apache.metron.common.dsl.GrammarUtils;
+import org.apache.metron.common.dsl.ParseException;
+import org.apache.metron.common.dsl.Token;
+import org.apache.metron.common.utils.ConversionUtils;
+import org.apache.metron.profiler.client.window.generated.WindowBaseListener;
+import org.apache.metron.profiler.client.window.generated.WindowLexer;
+import org.apache.metron.profiler.client.window.generated.WindowParser;
+import org.apache.metron.profiler.client.window.predicates.DayPredicates;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+public class WindowProcessor extends WindowBaseListener {
+  private Throwable throwable;
+  private Deque<Token<?>> stack;
+  private static final Token<Object> LIST_MARKER = new Token<>(null, Object.class);
+  private static final Token<Object> DAY_SPECIFIER_MARKER = new Token<>(null, Object.class);
+  private Window window;
+
+  public WindowProcessor() {
+    this.stack = new ArrayDeque<>();
+    this.window = new Window();
+  }
+
+  public Window getWindow() {
+    return window;
+  }
+
+  private void enterList() {
+    stack.push(LIST_MARKER);
+  }
+
+  private List<Function<Long, Predicate<Long>>> getPredicates() {
+    LinkedList<Function<Long, Predicate<Long>>> predicates = new LinkedList<>();
+    while (true) {
+      Token<?> token = stack.pop();
+      if (token == LIST_MARKER) {
```

Comment:
I should say, it's a specific static instance used for the parser to denote the end of the list of values in the stack.